src/query/storage/promremote/storage_test.go

```diff
@@ -248,6 +249,28 @@ func TestWriteBasedOnRetention(t *testing.T) {
 		assert.Contains(t, err.Error(), "test err")
 		assert.NotNil(t, promLongRetention2.GetLastWriteRequest())
 	})
+
+	t.Run("wrap non 5xx errors as invalid params error", func(t *testing.T) {
+		reset()
+		promLongRetention.SetError("test err", http.StatusForbidden)
+		err := sendWrite(storagemetadata.Attributes{
+			Resolution: 10 * time.Minute,
+			Retention:  8760 * time.Hour,
+		})
+		require.Error(t, err)
+		assert.True(t, xerrors.IsInvalidParams(err))
+	})
+
+	t.Run("429 should not be wrapped as invalid params", func(t *testing.T) {
```

Comment:
The test function is called `TestWriteBasedOnRetention`, but the two new tests actually look into error responses/status codes. I think they should be in a separate test function, e.g. `TestStatusCodeHandling` or similar.